### PR TITLE
Added TextField.onTap()

### LINKED ISSUE #15882: How to add on tap gesture on a textField?
## Add onTap gesture on a textField
onTap gesture using GestureDetector class is working for every element except for the textField type.
How can i use onTap gesture on textfield element?

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes https://github.com/flutter/flutter/issues/15882

### CODE REVIEW

File: packages/flutter/lib/src/material/text_field.dart [hunk 1]
```diff
@@ -298,6 +334,26 @@ class TextField extends StatefulWidget {
...
+  ///
+  /// If the textfield is created with [enabled] false, taps will not be
+  /// recognized.
+  ///
+  /// To be notified when the textfield gains or loses the focus, provide a
+  /// [focusNode] and add a listener to that.
+  ///
+  /// To listen to arbitrary pointer events without competing with the
+  /// textfield's internal gesture detector, use a [Listener].
+  final GestureTapCallback onTap;
```
Comment (xster) [inline packages/flutter/lib/src/material/text_field.dart]: This will be called twice in a double tap. Is that ok?

Comment (HansMuller) [reply packages/flutter/lib/src/material/text_field.dart]: I haven't looked at the "next PR" yet but if we're timing taps to detect double taps, then we don't need to call this more than once for a series of taps?

**Comment by xster** [comment]
LGTM

File: packages/flutter/lib/src/material/text_field.dart [hunk 2]
```diff
@@ -144,9 +145,44 @@ class TextField extends StatefulWidget {
...
+  /// ```
   ///
   /// If null, this widget will create its own [FocusNode].
+  ///
+  /// ## Keyboard
+  ///
+  /// Requesting the focus will typically cause the the keyboard to be shown
+  /// if it's not showing already.
+  ///
+  /// On Android, the user can hide the keyboard - withouth changing the focus -
```
Comment (Hixie) [inline packages/flutter/lib/src/material/text_field.dart]: typo: `withouth`

File: packages/flutter/lib/src/material/text_field.dart [hunk 3]
```diff
@@ -298,6 +334,26 @@ class TextField extends StatefulWidget {
   /// {@macro flutter.widgets.editableText.enableInteractiveSelection}
   final bool enableInteractiveSelection;
 
+  /// Called when the user taps on this textfield.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/text_field.dart]: nit: looks like we call it a "text field" (with a space) in the class docs

File: packages/flutter/lib/src/material/text_field.dart [hunk 4]
```diff
@@ -315,6 +372,7 @@ class TextField extends StatefulWidget {
     properties.add(IntProperty('maxLines', maxLines, defaultValue: 1));
     properties.add(IntProperty('maxLength', maxLength, defaultValue: null));
     properties.add(FlagProperty('maxLengthEnforced', value: maxLengthEnforced, ifTrue: 'max length enforced'));
+    properties.add(DiagnosticsProperty<GestureTapCallback>('onTap', onTap, defaultValue: false));
```
Comment (Hixie) [inline packages/flutter/lib/src/material/text_field.dart]: I don't think we usually add callbacks to debugFillProperties, fwiw. Normally if we want to expose that we add something like the list of gestures that are listened for (they have short names) or some such, but in this case it's probably easier to just not mention it at all. Not a big deal though.

**Comment by HansMuller** [comment]
@Hixie - updated TextField.dart per you feedback in https://github.com/flutter/flutter/pull/26088

**Comment by calvinmuller** [comment]
CupertinoTextField didn't get this love?

**Comment by HansMuller** [comment]
@calvinmuller can you open an issue for CupertinoTextField

**Comment by aurbina83** [comment]
> CupertinoTextField didn't get this love?

I just arrived here for the same issue. No CupertinoTextField love

**Pull request merged**